osu.Game/Overlays/OverlayScrollContainer.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.Containers;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays
+{
+    /// <summary>
+    /// <see cref="OsuScrollContainer"/> which provides <see cref="ScrollToTopButton"/>. Mostly used in <see cref="FullscreenOverlay"/>.
+    /// </summary>
+    public class OverlayScrollContainer : OsuScrollContainer
+    {
+        /// <summary>
+        /// Scroll position at which the <see cref="ScrollToTopButton"/> will be shown.
+        /// </summary>
+        private const int button_scroll_position = 200;
+
+        public ScrollToTopButton Button { get; }
+
+        private float currentTarget;
+
+        public OverlayScrollContainer()
+        {
+            AddInternal(Button = new ScrollToTopButton
+            {
+                Anchor = Anchor.BottomRight,
+                Origin = Anchor.BottomRight,
+                Margin = new MarginPadding(20),
+                Action = () =>
+                {
+                    ScrollToStart();
+                    currentTarget = Target;
+                    Button.Current.Value = Visibility.Hidden;
+                }
+            });
+        }
+
+        protected override void UpdateAfterChildren()
+        {
+            base.UpdateAfterChildren();
+
+            if (ScrollContent.DrawHeight + button_scroll_position < DrawHeight)
+            {
+                Button.Current.Value = Visibility.Hidden;
+                return;
+            }
+
+            if (Target == currentTarget)
+                return;
+
+            currentTarget = Target;
+            Button.Current.Value = Current > button_scroll_position ? Visibility.Visible : Visibility.Hidden;
+        }
+
+        public class ScrollToTopButton : OsuHoverContainer, IHasCurrentValue<Visibility>
+        {
+            private const int fade_duration = 500;
+
+            private readonly BindableWithCurrent<Visibility> current = new BindableWithCurrent<Visibility>();
+
+            public Bindable<Visibility> Current
```

Comment:
remove both bindable and just use a local backed property

```csharp
        private Visibility state;

        public Visibility State
        {
            get => state;
            set
            {
                if (value == state)
                    return;

                state = value;
                // do thing here          
            }
        }
```

Reply:
Done